task_sdk/tests/execution_time/test_task_runner.py

```diff
@@ -26,13 +26,47 @@
 import pytest
 from uuid6 import uuid7
 
+from airflow.exceptions import AirflowSkipException
 from airflow.sdk import DAG, BaseOperator
 from airflow.sdk.api.datamodels._generated import TaskInstance, TerminalTIState
 from airflow.sdk.execution_time.comms import DeferTask, SetRenderedFields, StartupDetails, TaskState
-from airflow.sdk.execution_time.task_runner import CommsDecoder, parse, run, startup
+from airflow.sdk.execution_time.task_runner import CommsDecoder, RuntimeTaskInstance, parse, run, startup
 from airflow.utils import timezone
 
 
+def mocked_parse(what: StartupDetails, dag, task_id) -> RuntimeTaskInstance:
+    """A mock parse function to support in line DAGs."""
+    # Retrieve the task from the DAG
+    task = dag.task_dict[task_id]
+    return RuntimeTaskInstance.model_construct(**what.ti.model_dump(exclude_unset=True), task=task)
+
+
+@pytest.fixture
+def mocked_parse_fixture(spy_agency):
+    def set_dag(what: StartupDetails, dag_id: str, task: BaseOperator):
+        dag = get_inline_dag(dag_id, task)
+        spy_agency.spy_on(parse, call_fake=lambda what: mocked_parse(what, dag, task.task_id))
+        return mocked_parse(what, dag, task.task_id)
```

Comment:
Worth just moving all logic of `mocked_parse` here so you can call this function `mocked_parse` instead of `mocked_parse_fixture`

Reply:
Then you can add description about when to use this fixture in docstrings

Reply:
Good idea, made the change